src/Integration/Binding/CSharpVBBindingOperation.cs

```diff
@@ -119,6 +125,23 @@ public void Commit()
             }
         }
 
+        private void AddAdditionalFile()
+        {
+            try
+            {
+                var projectSystem = serviceProvider.GetService<IProjectSystemHelper>();
+                projectSystem.AssertLocalServiceIsNotNull();
+                // It's possible that the additional file is set property, but the ruleset is out of date, which will trigger the binding. So we need to remove the file first, otherwise we get an error that the file already exists.
+                // It's possible that the additional file exists but has a wrong type, e.g. Compile/Content, and not AdditionalFiles. In which case removing and re-adding it solves the problem.
+                projectSystem.RemoveFileFromProject(initializedProject, cSharpVBBindingConfig.AdditionalFile.Path);
```

Comment:
Thinking about it again, this might causes a problem in practice. What will happen if the additional file is referenced via an imported targets file?
Even if it isn't, removing and re-adding the same reference might change where the line appears in the project file. This is less serious, but it's still irritating for end users to have unnecessary changes appearing when they diff the project file.

I'll see what the behaviour is when testing - it might be something we'll fix later.